packages/core/src/browser/label-provider.ts

```diff
@@ -106,6 +143,20 @@ export namespace URIIconReference {
 @injectable()
 export class DefaultUriLabelProviderContribution implements LabelProviderContribution {
 
+    protected formatters: ResourceLabelFormatter[] = [];
+    protected readonly onDidChangeEmitter = new Emitter<DidChangeLabelEvent>();
+    protected homePath: string | undefined;
+    constructor(@inject(EnvVariablesServer) protected readonly envVariablesServer: EnvVariablesServer) {
+        const setHomePath = (variable: EnvVariable | undefined) => {
+            if (variable && variable.value) {
+                this.homePath = variable.value;
+            }
+        };
+        this.envVariablesServer.getValue('HOME').then(variable => setHomePath(variable));
```

Comment:
let's wait for new fs API PR, and then use `enVariableServer.userDirUri` instead

Reply:
Did you mean `envVariablesServer.getHomeDirUri()`? or is there another PR?

Reply:
yes `getHomeDirUri`, sorry